providers/component-protocol/cpregister/base/provider_name.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package base
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	componentProviderNamePrefix        = "component-protocol.components."
+	defaultComponentProviderNamePrefix = "component-protocol.default-components."
+	defaultComponentProviderNamespace  = "default-components"
+)
+
+// MustGetScenarioAndCompNameFromProviderKey .
+func MustGetScenarioAndCompNameFromProviderKey(providerKey string) (scenario, compName, instanceName string) {
+	scenario, compName, instanceName, err := GetScenarioAndCompNameFromProviderKey(providerKey)
+	if err != nil {
+		panic(err)
+	}
+
+	return scenario, compName, instanceName
+}
+
+// GetScenarioAndCompNameFromProviderKey .
+func GetScenarioAndCompNameFromProviderKey(providerKey string) (scenario, compName, instanceName string, err error) {
+	if strings.HasPrefix(providerKey, defaultComponentProviderNamePrefix) {
+		ss := strings.SplitN(providerKey, ".", 3)
+		return "", ss[2], ss[2], nil
+	}
+	if !strings.HasPrefix(providerKey, componentProviderNamePrefix) {
+		return "", "", "", fmt.Errorf("invalid prefix")
+	}
+	ss := strings.SplitN(providerKey, ".", 4)
+	if len(ss) != 4 {
+		return "", "", "", fmt.Errorf("not standard provider key: %s", providerKey)
+	}
```

Comment:
should encapuslate a func like: remove prefix or what?

Reply:
adjusted